infocmdb/attribute.go

```diff
@@ -336,3 +338,282 @@ func (c *Client) GetCiAttributeValueCi(ciId int, attributeName string) (value st
 func (c *Client) UpdateCiAttribute(ci int, ua []v2.UpdateCiAttribute) (err error) {
 	return c.v2.UpdateCiAttribute(ci, ua)
 }
+
+type AttributeType int
+
+const (
+	AT_INPUT AttributeType = iota + 1
+	AT_TEXTAREA
+	AT_TEXTEDIT
+	AT_SELECTFIELD
+	AT_CHECKBOX
+	AT_RADIO
+	AT_DATE
+	AT_DATETIME
+	AT_ZAHLUNGSMITTEL
+	AT_PASSWORD
+	AT_LINK
+	AT_ATTACHMENT
+	AT_SCRIPT
+	AT_EXECUTEABLE
+	AT_QUERY
+	AT_CITYPE
+	AT_INFO
+	AT_QUERYPERSIST
+	AT_CITYPEPERSIST
+	AT_FILTER
+	AT_SELECTQUERY
+	AT_SELECTPOPUP
+)
+
+type Columns int
+
+const (
+	C_ONE Columns = iota + 1
+	C_TWO
+)
+
+type Multiselect int
+
+const (
+	MS_ZERO Multiselect = iota
+	MS_ONE
+	MS_TWO
+)
+
+type respCreateAttribute struct {
+	Success bool         `json:"success"`
+	Message string       `json:"message"`
+	Data    []responseId `json:"data"`
+}
+
+type AttributeParams struct {
+	Name                string
+	Description         string
+	Note                string
+	Hint                string
+	AttributeType       AttributeType
+	AttributeGroupName  string
+	OrderNumber         int
+	Column              Columns //values in db enum('1','2')
+	IsUnique            bool
+	IsNumeric           bool
+	IsBold              bool
+	IsEvent             bool
+	IsUniqueCheck       bool
+	IsAutocomplete      bool
+	IsMultiselect       Multiselect //values in db enum('0','1','2')
+	IsProjectRestricted bool
+	Regex               string
+	ScriptName          string
+	InputMaxlength      int
+	TextareaCols        int
+	TextareaRows        int
+	IsActive            bool
+	Historicize         bool
+	UserId              int
+}
+
+func (c *Client) NewAttributeParams() (params *AttributeParams) {
+	params = &AttributeParams{
+		AttributeType: 0,
+		Column:        C_ONE,
+		IsMultiselect: MS_ZERO,
+		IsActive:      false,
+		Historicize:   true,
+		UserId:        0,
+	}
+	return
+}
+
+func (c *Client) CreateAttribute(attributeParams *AttributeParams) (attributeId int, err error) {
+
+	if err = c.v2.Login(); err != nil {
+		return
+	}
+
+	existingAttributeId, err := c.GetAttributeIdByAttributeName(attributeParams.Name)
+	if err != nil && strings.Contains(err.Error(), "query returned no result") == false {
+		return 0, err
+	}
+
+	if existingAttributeId == 0 {
+
+		columns := []string{
+			"name",
+			"description",
+			"note",
+			"hint",
+			"attribute_type_id",
+			"attribute_group_id",
+			"order_number",
+			"column",
+			"is_unique",
+			"is_numeric",
+			"is_bold",
+			"is_event",
+			"is_unique_check",
+			"is_autocomplete",
+			"is_multiselect",
+			"is_project_restricted",
+			"regex",
+			"script_name",
+			"input_maxlength",
+			"textarea_cols",
+			"textarea_rows",
+			"is_active",
+			"user_id",
+			"historicize",
+		}
+
+		AttributeGroupId, err := c.GetAttributeGroupIdByName(attributeParams.AttributeGroupName)
```

Comment:
This private function scoped variable should start with a lower case letter.
```suggestion
		attributeGroupId, err := c.GetAttributeGroupIdByName(attributeParams.AttributeGroupName)
```

Reply:
Done.